src/graph_export/embeddings_graph_cli_parser.cpp

```diff
@@ -0,0 +1,142 @@
+//*****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "embeddings_graph_cli_parser.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include "../capi_frontend/server_settings.hpp"
+#include "../ovms_exit_codes.hpp"
+#include "../version.hpp"
+#include "../status.hpp"
+
+namespace ovms {
+
+EmbeddingsGraphSettingsImpl& EmbeddingsGraphCLIParser::defaultGraphSettings() {
+    static EmbeddingsGraphSettingsImpl instance;
+    return instance;
+}
+
+void EmbeddingsGraphCLIParser::createOptions() {
+    this->options = std::make_unique<cxxopts::Options>("ovms --pull [PULL OPTIONS ... ]", "-pull --task embeddings graph options");
+
+    // clang-format off
+    options->add_options("embeddings")
+        ("graph_target_device",
+            "CPU, GPU, NPU or HETERO, default is CPU.",
+            cxxopts::value<std::string>()->default_value("CPU"),
+            "GRAPH_TARGET_DEVICE")
+        ("num_streams",
+            "The number of parallel execution streams to use for the model. Use at least 2 on 2 socket CPU systems.",
+            cxxopts::value<uint32_t>()->default_value("1"),
+            "NUM_STREAMS")
+        ("truncate",
+            "Truncate the prompts to fit to the embeddings model.",
+            cxxopts::value<std::string>()->default_value("false"),
+            "TRUNCATE")
+        ("normalize",
+            "Normalize the embeddings.",
+            cxxopts::value<std::string>()->default_value("false"),
+            "NORMALIZE")
+        ("version",
+            "Version of the model.",
```

Comment:
Is it version of exported model from HF? Is it directory number version?

Reply:
This is currently not used, but will be used once we also export models from optimum cli. Addedto match model_export.py parameters.